### PR TITLE
Use qLog variants in `qehvi_candidates_func` and `qnehvi_candidates_func`

### LINKED ISSUE #299: Support for qLogExpectedHypervolumeImprovement
### Expected behavior

I'm using optuna-integration.botorch to run a 6 parameter/2 objective study. After the startup trials, I get the following warning:

```
botorch/acquisition/multi_objective/monte_carlo.py:110: NumericsWarning: qExpectedHypervolumeImprovement has known numerical issues that lead to suboptimal optimization performance. It is strongly recommended to simply replace

         qExpectedHypervolumeImprovement         -->     qLogExpectedHypervolumeImprovement

instead, which fixes the issues and has the same API. See https://arxiv.org/abs/2310.20708 for details.
  legacy_ei_numerics_warning(legacy_name=type(self).__name__)
```

I don't see anywhere the Log version is used for HypervolumeImprovement yet.

### Environment

- Optuna version: 4.9.0
- Optuna Integration version: 4.9.0
- Python version: 3.14
- OS: WSL Ubuntu
- (Optional) Other libraries and their versions: botorch version 0.18.1

### Error messages, stack traces, or logs

```shell
botorch/acquisition/multi_objective/monte_carlo.py:110: NumericsWarning: qExpectedHypervolumeImprovement has known numerical issues that lead to suboptimal optimization performance. It is strongly recommended to simply replace

         qExpectedHypervolumeImprovement         -->     qLogExpectedHypervolumeImprovement

instead, which fixes the issues and has the same API. See https://arxiv.org/abs/2310.20708 for details.
  legacy_ei_numerics_warning(legacy_name=type(self).__name__)
```

### Steps to reproduce

1. Use optuna.create_study to create a study using BoTorchSampler which would go down the qExpectedHypervolumeImprovement path. You should get the warning about switching to qLogExpectedHypervolumeImprovement

### Additional context (optional)

_No response_

### PR DESCRIPTION
**Pull request by adrianhtt**
## Motivation

I was running a 2 objective study with BoTorchSampler and every trial after the startup phase printed this:

```
NumericsWarning: qExpectedHypervolumeImprovement has known numerical issues that lead to suboptimal optimization performance. It is strongly recommended to simply replace

         qExpectedHypervolumeImprovement         -->     qLogExpectedHypervolumeImprovement

instead, which fixes the issues and has the same API. See https://arxiv.org/abs/2310.20708 for details.
```

The recommendation comes from the LogEI paper (https://arxiv.org/abs/2310.20708), so this is not just warning noise, the legacy class can genuinely underperform.

It was annoying enough that I went digging. Interestingly this was already supposed to be fixed: #242 added a fallback that prefers the qLog variant when it is available. The catch is that it looks for the class on `botorch.acquisition.multi_objective.monte_carlo`, and botorch actually keeps the qLog variants in `botorch.acquisition.multi_objective.logei`. As far as I can tell the class was never importable from monte_carlo in any botorch release, so the hasattr check never succeeds and the legacy qEHVI is silently used every time. `qnehvi_candidates_func` has the same problem in a simpler form, it never had a qLog path at all.

Closes #299.

## Description of the changes

- Added a try_import block for `qLogExpectedHypervolumeImprovement` and `qLogNoisyExpectedHypervolumeImprovement`, following the existing `_imports_qlogei` pattern
- `qehvi_candidates_func` and `qnehvi_candidates_func` use the qLog variants when the import succeeds and fall back to the legacy classes on older botorch
- Added a regression test that asserts the qLog acquisition class is actually used and that no numerical issue warning is emitted

Verified on botorch 0.9.5 and 0.18.1: a 2 objective study produced 3 NumericsWarnings before this change and none after, on both the qehvi and qnehvi paths.

### CODE REVIEW

**Comment by adrianhtt** [comment]
The CI failure on 3.10 to 3.13 turned out to be more interesting than a flake. With 7 objectives and only a few observed points, the approximate box decomposition (alpha > 0) can prune every cell, and qLogEHVI raises an IndexError on an empty decomposition while the legacy qEHVI just returns zero improvement. I could reproduce it with a fixed seed on both botorch 0.9.5 and 0.18.1, so it is not something a version check can solve.

I pushed a fallback: qehvi_candidates_func now only picks qLogEHVI when the decomposition has at least one cell, plus a deterministic regression test for the empty case. The 3.9 job passed before only because old botorch does not have the qLog module at all, so it was already on the legacy path.

The empty decomposition crash itself is probably worth reporting to botorch upstream separately.

File: tests/botorch/test_botorch.py [hunk 1]
```diff
@@ -149,6 +149,79 @@ def test_botorch_specify_candidates_func(candidates_func: Any, n_objectives: int
...
+        [
+            torch.zeros(n_objectives, dtype=torch.double),
+            torch.ones(n_objectives, dtype=torch.double),
+        ]
+    )
+
+    ref_point = train_obj.min(dim=0).values - 1e-8
+    partitioning = NondominatedPartitioning(ref_point=ref_point, Y=train_obj, alpha=0.1)
+    if partitioning.get_hypercell_bounds().shape[-2] != 0:
+        pytest.skip("This seed did not produce an empty box decomposition on this platform.")
```
Comment (y0z) [inline tests/botorch/test_botorch.py]: Using a patch can guarantee an empty box decomposition without randomness.
```python
with patch.object(
    NondominatedPartitioning,
    "get_hypercell_bounds",
    return_value=torch.zeros(2, 0, n_objectives, dtype=torch.double),
):
    candidates = integration.botorch.qehvi_candidates_func(
        train_x, train_obj, None, bounds, None
    )
    assert torch.isfinite(candidates).all()
```

**Review by y0z — COMMENTED** [review]
Thank you for your PR.

The sampler changes are LGTM. Please fix the test based on my comment.

File: tests/botorch/test_botorch.py (hunk 1 above)
Comment (adrianhtt) [reply tests/botorch/test_botorch.py]: Good idea, that is much more robust than relying on the seed. Updated the test to use the patch. Thanks for the review!

**Review by y0z — APPROVED** [review]
LGTM
